Pages/Pag81751.NSPPerformanceTest.al

```diff
@@ -152,119 +195,177 @@ page 81751 "NSP Performance Test"
         CreateNumberSeries();
     end;
 
-    local procedure InitEntryTable()
+    local procedure TestNumberIncrement(ShowMessage: Boolean; BulkInsert: Boolean)
     var
-        EntryTable: Record "NSP Entry Table";
-        EntryTable2: Record "NSP Entry Table2";
         StartDateTime: DateTime;
         i: Integer;
+        Message: text[250];
+        StartSqlStatements, StartSqlRowsRead : BigInteger;
     begin
-        EntryTable.DeleteAll();
-        EntryTable2.DeleteAll();
+        DeleteEntries();
+        StartSqlStatements := SessionInformation.SqlStatementsExecuted;
+        StartSqlRowsRead := SessionInformation.SqlRowsRead;
+        if BulkInsert then begin
+            Message := 'Number Inserts with Bulk inserts'
+        end else
+            Message := 'Number Inserts without Bulk inserts';
+
+
         StartDateTime := CurrentDateTime();
 
         for i := 1 to NumberOfIterations do begin
-            EntryTable.Init();
-            EntryTable."Entry No." := i;
-            EntryTable.Description := 'test';
-            EntryTable.Insert(true);
+            CreateEntry(i, BulkInsert);
```

Comment:
I understood that which is a valid point. However you had had scenarios where you actually inserted records. So I added it to the other places so we would not compare apples with pears.  Also the tables are very simple with not that many keys and reality i production environments we also have to factor that as well.

Another factor to this that the thing I'm working in is helping optimizing an ISV that is ~20 years old that has a lot of findlast +1 entries and this gives terrible locking for the customer case with 600 licenses. 
So that is why I'm VERY interested on the different methods of getting a unique no when inserting records.

You are right about the actions. The purpose of the beginning was not to do a PR but to tweak it so I could verify the difference between SQL and Azure SQL plus some other tests.
Have you seen Waldos Techdays session "Coding for performance"? He developed a framework for testing different "performance test" and showing the result side by side.

Reply:
I only have one test where Insert occurs is property AutoIncrement, as it requires a Record to test it. And in the article I stated that "Unfortunately the Auto Increment property can't be measured in isolation from Insert so I'll calculate this roughly using the difference between inserting 100.000 records with Auto Increment and without." So, in article result is (**Insert100.000RecordsWithAutoIncrement-Insert100.000Records**)

Just because the tables can be quite different they should be excluded from the measurements, if we are just talking about number generation.

FindLast with Insert is a very valid point. So I think your tests are also very important and necessary! So I will gladly add them, but without breaking the logic of my tests according to released blog posts :)

I understand your train of thought when you made these changes and I apologize for my perfectionism. I can't bring myself to treat code in master more easily, even if it's temporary code for testing :D

Reply:
@NAVFreak 
Yes I saw the Waldo session, but not very closely. I don't remember if his framework is free or what it is

Reply:
Do what you want with my code. I didn't expect you to accept the PR due to the massive "rewriting". Copy the parts you like 😊

His repo is here: https://github.com/waldo1001/waldo.BCPerfTool
And the session: https://www.youtube.com/watch?v=E3ADZsisFbE
I have added some stuff to it as well.